BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsBaseWorker.kt

```diff
@@ -0,0 +1,23 @@
+package com.braintreepayments.api
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+
+// NEXT_MAJOR_VERSION: remove this class, it may no longer be needed
+
+/**
+ * Class for background analytics tasks.
+ * This class is used internally by the SDK and should not be used directly.
+ * It is not subject to semantic versioning and may change at any time.
+ * @suppress
+ */
+abstract class AnalyticsBaseWorker(
+    context: Context,
+    workerParams: WorkerParameters
+) : Worker(context, workerParams) {
+
+    internal open fun createAnalyticsClientFromInputData(): AnalyticsClient {
```

Comment:
this method doesnt need the `open` access modifier. Its is an internal method used only in subclasses `AnalyticsWriteToDbWorker` and `AnalyticsUploadToDbWorker`. I think we are safe to delete `open` modifiers for classes that are for internal use of the SDK (even if they are public), as they are not part of semantic versioning

Reply:
I agree overall, but I would still defer removing `open` until the next major version to follow strict semantic versioning rules. Technically a user can subclass `AnalyticsBaseWorker` currently, and they should be able to subclass this method since it's marked as `protected` on the Java side and Kotlin doesn't have an equivalent to protected.

To be totally clear I don't _want_ to do this lol, but I'm following the rules of semantic versioning and the mantra of "If it can happen, it will happen." If it's possible now, then we should support the 0.000001% of SDK users that do this. Then we can tighten the restrictions in the next major version. It's counter intuitive but it leads to high quality SDKs.

Reply:
@sshropshire makes perfect sense!

Reply:
@jcnoriega actually you're right on this one. I just checked a Java 7 Demo I made. `AnalyticsClient` is marked as internal so a merchant won't be able to declare a subclass.

Some of the Java / Kotlin conversion is gonna be messy for a bit. I want to clean it up at some point, but I do want to continue being thorough with semantic versioning whenever possible.